pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1759,18 +1732,67 @@ func appendKubernetesRemoteLocation(xsite *ispnutil.XSite, infinispan *infinispa
 	return nil
 }
 
-func getRemoteSiteServiceHostPort(service *corev1.Service, remoteKubernetes *ispnutil.Kubernetes, logger logr.Logger) (string, int32, error) {
+func getCrossSiteServiceHostPort(service *corev1.Service, kubernetes *ispnutil.Kubernetes, logger logr.Logger) (string, int32, error) {
 	switch serviceType := service.Spec.Type; serviceType {
 	case corev1.ServiceTypeNodePort:
 		// If configuring NodePort, expect external IPs to be configured
-		return remoteKubernetes.PublicIP(), remoteKubernetes.GetNodePort(service), nil
+		return getNodePortServiceHostPort(service, kubernetes, logger)
 	case corev1.ServiceTypeLoadBalancer:
 		return getLoadBalancerServiceHostPort(service, logger)
 	default:
 		return "", 0, fmt.Errorf("unsupported service type '%v'", serviceType)
 	}
 }
 
+func getNodePortServiceHostPort(service *corev1.Service, k *ispnutil.Kubernetes, logger logr.Logger) (string, int32, error) {
+	//The NodePort can be found in the service
+	port := service.Spec.Ports[0].NodePort
+
+	//The IPs must be fetch. Some cases, the API server (which handles REST requests) isn't the same as the worker
+	//So, we get the workers list. It needs some permissions cluster-reader permission
+	//oc create clusterrolebinding <name> -n ${NAMESPACE} --clusterrole=cluster-reader --serviceaccount=${NAMESPACE}:<account-name>
+	workerList := &corev1.NodeList{}
+
+	//select workers only. AFAIK, only the workers have the proxy to the pods
+	labelSelector := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/worker": ""})
```

Comment:
@pruivo do we also need to check the Status of the Node. Is node in any status and state in enough?

Reply:
I wasn't worried about it for 2 reasons:
1. every blog post I read about NodePort says it isn't recommended in production. 
2. This is used for cross-site replication and our image doesn't support more than an <ip:port> pair per site. So, if the node crashes, it is going to fail and the Infinispan deployment must be deleted and re-applied on all sites.

Reply:
Yes, @pruivo true about NodePort
But you are listing all worker Nodes and takes the first one, but it can be in NotReady state by origin. Is it correct? You are not validating state of the node.

Reply:
@dmvolod I've updated in my PR: https://github.com/infinispan/infinispan-operator/pull/441
@rigazilla do you want to merge my PR separately or cherry-pick the commit to yours?

Reply:
If it ok to you, I think we can go faster with a cherry pick.

Reply:
ok. it works for me.